processors/ratelimiter/ratelimiterprocessor.go

```diff
@@ -0,0 +1,156 @@
+package ratelimiter
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	pb_struct "github.com/envoyproxy/go-control-plane/envoy/extensions/common/ratelimit/v3"
+	pb "github.com/envoyproxy/go-control-plane/envoy/service/ratelimit/v3"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+var _ component.TracesProcessor = (*processor)(nil)
+
+var (
+	tagTenantID               = tag.MustNewKey("tenant-id")
+	droppedSpanCountSoftLimit = stats.Int64("tenant_id_dropped_span_count_soft_limit", "Number of spans dropped because of rate limiting per tenant due to soft limit", stats.UnitDimensionless)
+	droppedSpanCount          = stats.Int64("tenant_id_dropped_span_count", "Number of spans dropped per tenant due to global rate limiting", stats.UnitDimensionless)
+)
+
+func MetricViews() []*view.View {
```

Comment:
Put this metrics logic in a separate go file. Called metrics.go or something like that.

Reply:
done